src/packagedcode/rubygems.py

```diff
@@ -648,92 +628,172 @@ def normalize(gem_data, known_fields=known_fields):
 
 
 def parse_spec(location):
-    pass
+    """
+    Returns a dictionary containing .gemspec file data.
+    """
+    file_data = []
+    with io.open(location, "r", encoding="utf-8") as f:
+        file_data = f.read().splitlines()
+
+    dependencies = OrderedDict()
```

Comment:
Why do we define this way up here? If we want to define an empty dict, we should do it closer to where its actually used in code.

Reply:
I have used this dictionary in for loop. In python2 and python3  normal dictionary was stores data in different ways so I used this here to store in a particular order only.